openapi-core/src/main/java/io/ballerina/openapi/core/generators/schema/BallerinaTypesGenerator.java

```diff
@@ -65,9 +66,24 @@
  */
 public class BallerinaTypesGenerator {
 
-    private List<TypeDefinitionNode> typeDefinitionNodeList;
+    private List<TypeDefinitionNode> typeDefinitionNodeList = new LinkedList<>();
     private boolean hasConstraints;
 
+    /**
+     * This public constructor is used to generate record and other relevant data type when the nullable flag is
+     * enabled in the openapi command.
+     *
+     * @param openAPI    OAS definition
+     * @param isNullable nullable value
+     * @param  typeDefinitionNodeList list of types generated by earlier generations
+     */
+    public BallerinaTypesGenerator(OpenAPI openAPI, boolean isNullable,
+                                   List<TypeDefinitionNode> typeDefinitionNodeList) {
+        GeneratorMetaData.createInstance(openAPI, isNullable);
+        this.typeDefinitionNodeList = typeDefinitionNodeList;
+        this.hasConstraints = false;
+    }
```

Comment:
Can we modify all the overloaded constructors to use one another, instead of duplicating the same constructor bodies?